social_core/backends/azuread.py

```diff
@@ -79,8 +79,8 @@ def access_token_url(self):
         return self.ACCESS_TOKEN_URL.format(base_url=self.base_url)
 
     def get_user_id(self, details, response):
-        """Use upn as unique id"""
-        return response.get("upn")
+        """Use subject (sub) claim as unique id."""
+        return response.get("sub")
```

Comment:
This changes `uid` for existing installation what would be a breaking change. Why `upn` is not present in your claim?